src/Approximations/overapproximate.jl

```diff
@@ -1148,9 +1143,22 @@ function _overapproximate_vTM_zonotope!(vTM, c, gen_lin, gen_rem)
         α = mid(rem_nonlin)
         c[i] = constant_term(Q) + α  # constant terms
         gen_lin[i, :] = get_linear_coeffs(Q) # linear terms
-        gen_rem[i] = abs(rem_nonlin.hi - α)
+        aux = abs(rem_nonlin.hi - α)
+        gen_rem[i] = aux
+        if !isapproxzero(aux)
+            nz_gen_rem += 1
+        end
+    end
+    if nz_gen_rem > 0
+        D = Diagonal(gen_rem)
+        if nz_gen_rem < size(D, 2)
+            D = D[:, 1:nz_gen_rem]
```

Comment:
This does not make sense to me. I would expect that you have to remove those entries that are zero (and not just the last ones)

Reply:
you're right, thanks. ill fix it in the loop

Reply:
oops, i think that this is still wrong because `gen_rem[nz_gen_rem] = aux` may not place the element in the correct *row*

Reply:
fixed -- i reverted these changes, and now drop the zeros in the `nonzero_columns` for sparse matrices